### PR TITLE
LPS-190337 - Check that document belongs to current company.

### PR DESCRIPTION
**Pull request by manuelRives**
As previously mentioned, if someone acceded a file stored in a virtual instance using a URL based on the main instance the file was accessible. With this check if the file does not belong to the current instance a NoSuchFileEntryException exception is returned.

@robertoDiaz also has the full context.

### CODE REVIEW

**Comment by liferay-lima** [comment]
ci:test:sf

**Comment by liferay-lima** [comment]
ci:test:relevant

File: portal-impl/src/com/liferay/portal/webserver/WebServerServlet.java [hunk 1]
```diff
@@ -1586,6 +1588,35 @@ private void _checkFileEntry(
...
+				_log.debug(
+					StringBundler.concat("Group not found: ", groupId, "."));
+			}
+
+			throw new NoSuchFileEntryException("No file entry found");
+		}
+
+		long companyId = PortalUtil.getCompanyId(httpServletRequest);
+
+		if (!(group.getCompanyId() == companyId)) {
```
Comment (adolfopa) [inline portal-impl/src/com/liferay/portal/webserver/WebServerServlet.java:1608]: Why not use `group.getCompanyId() != companyId` here?

**Review by adolfopa — CHANGES_REQUESTED** [review]

File: portal-impl/src/com/liferay/portal/webserver/WebServerServlet.java (hunk 1 above)
Comment (manuelRives) [reply portal-impl/src/com/liferay/portal/webserver/WebServerServlet.java:1608]: Completely agree, I must have changed it trying other conditions and I did not change it again

Comment (manuelRives) [reply portal-impl/src/com/liferay/portal/webserver/WebServerServlet.java]: Changed

**Comment by manuelRives** [comment]
ci:test:relevant

**Review by adolfopa — APPROVED** [review]

**Comment by adolfopa** [comment]
ci:forward

**Comment by WYZ-Issac** [comment]
Failures are unrelated. Forward directly.

**Comment by WYZ-Issac** [comment]
ci:forward:force

**Pull request closed without merging**